python/lsst/pipe/tasks/finalizeCharacterization.py

```diff
@@ -0,0 +1,693 @@
+            #   table_source[b] <-> table_cat[use_band[a]]
+            obj_index = table_source['obj_index'][:]
+            a, b = esutil.numpy_util.match(use_band, obj_index)
+
+            # Update indexes and cut to band-selected stars/sources
+            table_source['obj_index'][b] = a
+            _, index_new = np.unique(a, return_index=True)
+            table_cat[f'source_cat_index_{band}'][use_band] = index_new
+
+            table_source = table_source[b]
+            table_cat = table_cat[use_band]
+
+            # Add reserved flag column to tables
+            table_cat = np.lib.recfunctions.append_fields(
+                table_cat,
+                'reserved',
+                np.zeros(table_cat.size, dtype=bool),
+                usemask=False
+            )
+            table_source = np.lib.recfunctions.append_fields(
+                table_source,
+                'reserved',
+                np.zeros(table_source.size, dtype=bool),
+                usemask=False
+            )
+
+            # Get reserve star flags
+            table_cat['reserved'][:] = self.reserve_selection.run(
+                len(table_cat),
+                extra=f'{band}_{tract}',
+            )
+            table_source['reserved'][:] = table_cat['reserved'][table_source['obj_index']]
+
+            # Offset indices to account for tract merging
+            table_cat[f'source_cat_index_{band}'] += merge_source_counter
+            table_source['obj_index'] += merge_cat_counter
+
+            isolated_tables.append(table_cat)
+            isolated_sources.append(table_source)
+
+            merge_cat_counter += len(table_cat)
+            merge_source_counter += len(table_source)
+
+        isolated_table = np.concatenate(isolated_tables)
+        isolated_source_table = np.concatenate(isolated_sources)
+
+        return isolated_table, isolated_source_table
+
+    def _compute_psf_and_ap_corr_map(self, visit, detector, exposure, src, isolated_source_table):
+        """Compute psf model and aperture correction map for a single exposure.
+
+        Parameters
+        ----------
+        visit : `int`
+            Visit number (for logging).
+        detector : `int`
+            Detector number (for logging).
+        exposure : `lsst.afw.image.ExposureF`
+        src : `lsst.afw.table.SourceCatalog`
+        isolated_source_table : `np.ndarray`
+
+        Returns
+        -------
+        psf : `lsst.meas.algorithms.ImagePsf`
+            PSF Model
+        ap_corr_map : `lsst.afw.image.ApCorrMap`
+            Aperture correction map.
+        measured_src : `lsst.afw.table.SourceCatalog`
+            Updated source catalog with measurements, flags and aperture corrections.
+        """
+        # Apply source selector (s/n, flags, etc.)
+        good_src = self.source_selector.selectSources(src)
+
+        # Cut down input src to the selected sources
+        selection_mapper, selection_schema = self._make_selection_schema_mapper(src.schema)
```

Comment:
Instead of having an extra schema and `SchemaMapper` in play, it seems like you could simplify by creating the `measured_src` catalog here, and dropping rows from it that the `SourceSelector` didn't select.  But I'm guessing I'm missing something.

Reply:
With the mix of columns that are being rerun and not rerun, I could not make the single schema play work.  I did not have more patience to try fighting the system any more, and this seemed more reliable.

Reply:
:+1:, please just add a comment to that effect.